actions/postgresql_show_create_table_action.go

```diff
@@ -0,0 +1,469 @@
+}
+
+func (a *postgresqlShowCreateTableAction) printTableInit(ctx context.Context, w io.Writer, db *sql.DB) (string, error) {
+	var tableID, schema, relname string
+	row := db.QueryRowContext(ctx, `SELECT /* pmm-agent */  c.oid,
+	       n.nspname,
+	       c.relname
+	FROM pg_catalog.pg_class c
+	         LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+	WHERE c.relname = $1
+	  AND pg_catalog.pg_table_is_visible(c.oid)
+	ORDER BY nspname, relname;`, a.params.Table)
+	if err := row.Scan(&tableID, &schema, &relname); err != nil {
+		if err == sql.ErrNoRows {
+			return "", errors.Wrap(err, "Table not found")
+		}
+		return "", errors.WithStack(err)
+	}
+	fmt.Fprintf(w, "Table \"%s.%s\"\n", schema, relname) //nolint:errcheck
+	return tableID, nil
+}
+
+func (a *postgresqlShowCreateTableAction) sealed() {}
+
+func (a *postgresqlShowCreateTableAction) printColumnsInfo(ctx context.Context, w io.Writer, db *sql.DB, tableID string) error {
+	rows, err := db.QueryContext(ctx, `SELECT /* pmm-agent */ a.attname,
+       pg_catalog.format_type(a.atttypid, a.atttypmod),
+       (SELECT substring(pg_catalog.pg_get_expr(d.adbin, d.adrelid) for 128)
+        FROM pg_catalog.pg_attrdef d
+        WHERE d.adrelid = a.attrelid
+          AND d.adnum = a.attnum
+          AND a.atthasdef),
+       a.attnotnull,
+       (SELECT c.collname
+        FROM pg_catalog.pg_collation c,
+             pg_catalog.pg_type t
+        WHERE c.oid = a.attcollation
+          AND t.oid = a.atttypid
+          AND a.attcollation <> t.typcollation)                          AS attcollation,
+       ''::pg_catalog.char                                               AS attidentity,
+       a.attstorage,
+       CASE WHEN a.attstattarget = -1 THEN NULL ELSE a.attstattarget END AS attstattarget,
+       pg_catalog.col_description(a.attrelid, a.attnum)
+FROM pg_catalog.pg_attribute a
+WHERE a.attrelid = $1
+  AND a.attnum > 0
+  AND NOT a.attisdropped
+ORDER BY a.attnum;`, tableID)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer rows.Close()
+
+	tw := tabwriter.NewWriter(w, 0, 0, 1, ' ', tabwriter.Debug)
+
+	fmt.Fprintln(tw, "Column\tType\tCollation\tNullable\tDefault\tStorage\tStats target\tDescription") //nolint:errcheck
+
+	for rows.Next() {
+		var ci columnInfo
+		err = rows.Scan(
+			&ci.Attname,
+			&ci.FormatType,
+			&ci.Substring,
+			&ci.Attnotnull,
+			&ci.Attcollation,
+			&ci.Attidentity,
+			&ci.Attstorage,
+			&ci.Attstattarget,
+			&ci.ColDescription,
+		)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		fmt.Fprintf(tw, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
+			ci.Attname,
+			ci.FormatType,
+			pointer.GetString(ci.Attcollation),
+			formatNullable(ci.Attnotnull),
+			pointer.GetString(ci.Substring),
+			formatStorage(ci.Attstorage),
+			pointer.GetString(ci.Attstattarget),
+			pointer.GetString(ci.ColDescription),
+		) //nolint:errcheck
+	}
+	err = rows.Err()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	return tw.Flush()
```

Comment:
Technically, it is possible to have a PostgreSQL table with zero columns. Then we will have a header in the output, but zero rows.

But I don't think we have to cover that case. 😆